opengovsg/postmangovsg
# feat(frontend): styling for rich text editor popups

## Problem

Closes #1288
Closes #1305

## Solution

**Features**:
- Show placeholder for external images in rich text editor and preview
- Show placeholder for invalid images in rich text editor and preview
- Validate file.go.gov.sg image links

## Screenshots
![image](https://user-images.githubusercontent.com/3666479/129005581-87642134-97f5-480e-8f9e-31b4ce386299.png)

## Tests
- Insert text that is not a valid URL into the source field. An error message should be shown.
- Insert an image with a link from a non-whitelisted domain. The image should be inserted as a placeholder in the rich text editor. The image should also be a placeholder in the preview. The image width should adjust based on the value set.
- Insert an image with a link from a whitelisted domain. The image should be inserted correctly with link in both the editor and preview.
- Insert an image with a link from a whitelisted domain that is invalid (non-existent file.go.gov.sg link). An error message should be shown.
- Send out campaign with images. Emails should render correctly.
